mindsdb/integrations/utilities/time_series_utils.py

```diff
@@ -82,3 +86,61 @@ def get_best_model_from_results_df(nixtla_results_df, metric=r2_score):
         if accuracy > current_accuracy:
             best_model, current_accuracy = model, accuracy
     return best_model
+
+
+def spec_hierarchy_from_list(col_list):
+    """Gets the hierarchy spec from the list of hierarchy cols"""
+    spec = [["Total"]]
+    for i in range(len(col_list)):
+        spec.append(["Total"] + col_list[: i + 1])
+    return spec
+
+
+def get_hierarchy_from_df(df, model_args):
+    """Extracts hierarchy from the raw df, using the provided spec and args.
+
+    The "hierarchy" model arg is a list of format
+    [<level 1>, <level 2>, ..., <level n>]
+    where each element is a level in the hierarchy.
+
+    We return a tuple (nixtla_df, hier_df, hier_dict) where:
+    nixtla_df is a dataframe in the format nixtla packages uses for training
+    hier_df is a matrix of 0s and 1s showing the hierarchical structure
```

Comment:
I would suggest pointing the reader to the unit tests in order to understand what each of these looks like.

Reply:
Good shout!